app/taskroutes.py

```diff
@@ -0,0 +1,148 @@
+from flask import Blueprint, jsonify, abort, make_response, request
+from app.models.task import Task
+from app.models.goal import Goal
+from app import db
+from sqlalchemy import asc, desc
+import datetime, requests, os
+from dotenv import load_dotenv
+
+load_dotenv()
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except:
+        abort(make_response({"message":f"{cls.__name__} {model_id} invalid"}, 400))
+
+    model = cls.query.get(model_id)
+
+    if not model:
+        abort(make_response({"message":f"{cls.__name__} {model_id} not found"}, 404))
+
+    return model
+
+
+@tasks_bp.route("", methods=["GET"])
+def read_all_tasks():
+    title_query = request.args.get("title")
+    sort_query = request.args.get("sort")
+    
+    if title_query:
+        tasks = Task.query.filter_by(title=title_query)
+        
+    if sort_query == "asc":
+        tasks = Task.query.order_by(Task.title.asc())
+        
+    if sort_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+        
+    if not title_query and not sort_query:
+            
+        tasks = Task.query.all()
+    
+    
+    tasks_response = []
+    for task in tasks:
+        tasks_response.append(
+            {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": False 
+            }
+        )
+        
+    return jsonify(tasks_response)
+    
+
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def read_one_task(task_id):
+    task = validate_model(Task, task_id)
+    
+    if not task.goal_id:
+        return jsonify({"task": task.to_dict()}), 200
+
+    else:
+        return jsonify({
+            "task": {
+                "id": task.task_id,
+                "goal_id": task.goal_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": bool(task.completed_at)
+            }}), 200
+    
+
+
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+    if len(request_body) != 2:
+        return jsonify({"details": "Invalid data"}), 400
```

Comment:
This is ok for validation, but note that this only checks that there are two keys in the dictionary, not the names of those keys or if they have usable values. So if I was to send a body of:

```json
{
    "turkey": "Gobble gobble",
    "destruction": "Boom Boom"
}
```

This code will end up throwing a 500 error when it could have thrown a 400.